### DISCUSSION TITLE
How to deal with ORPHAN data type status

### DISCUSSION THREAD

**Discussion by gynt** [NationalSecurityAgency/ghidra#7184] [Q&A]
I have a project archive that I use to understand a program. For some reason, function signatures have been orphaned from the project source archive. If I click "Associate with archive" it says it is already associated. What is the best solution for this situation? I tried copy pasting them to the project archive, but that didn't do anything.

![image](https://github.com/user-attachments/assets/d45bd7c0-aa61-4216-8b24-f9aeab323ca1)

It says the project archive data types have been removed, but that isn't true, I can still see them when I browse for them.

![image](https://github.com/user-attachments/assets/7531384c-c952-4f90-87e5-cf120868295b)

Committing doesn't work, copy pasting doesn't work, exporting them to a c header doesn't actually export them, "Replacing" them with the prorgram's copy of the data type doesn't work.

**Comment by astrelsky** [comment]
I don't think this is actually helpful, but I usually just try to ignore them and pretend everything is fine.

**Comment by dragonmacher** [comment]
The help suggests you can simply perform a 'Commit' operation.

![image](https://github.com/user-attachments/assets/d9624040-74f2-44bc-9c3d-1a1702227923)

**Reply by gynt** [reply]
The problem is, the data types aren't actually deleted in the source archive, so the commit doesn't do anything it seems.
